scripts/ci-subtask.sh

```diff
@@ -29,11 +29,23 @@ else
     weight() {
         [[ $1 == "github.com/tikv/pd/server/api" ]] && return 30
         [[ $1 == "github.com/tikv/pd/pkg/schedule" ]] && return 30
-        [[ $1 == "pd/tests/server/api" ]] && return 30
+        [[ $1 == "github.com/tikv/pd/pkg/core" ]] && return 30
+        [[ $1 == "github.com/tikv/pd/tests/server/api" ]] && return 30
         [[ $1 =~ "pd/tests" ]] && return 5
         return 1
     }
 
+    # Create an associative array to store the weight of each task.
+    declare -A task_weights
+    for t in ${tasks[@]}; do
+        weight $t
+        task_weights[$t]=$?
+    done
+
+    # Sort tasks by weight in descending order.
+    IFS=$'\n' tasks=($(sort -rn <<<"${tasks[*]}"))
```

Comment:
What does IFS mean?

Reply:
https://unix.stackexchange.com/questions/184863/what-is-the-meaning-of-ifs-n-in-bash-scripting

Reply:
how about just `tasks=($(printf "%s\n" "${tasks[@]}" | sort -rn))`